src/js/game/systems/item_processor.js

```diff
@@ -189,22 +159,28 @@ export class ItemProcessorSystem extends GameSystemWithFilter {
      * @param {Entity} entity
      */
     canProcess(entity) {
+        const acceptorComp = entity.components.ItemAcceptor;
         const processorComp = entity.components.ItemProcessor;
 
         switch (processorComp.processingRequirement) {
             // DEFAULT
             // By default, we can start processing once all inputs are there
             case null: {
-                return processorComp.inputCount >= processorComp.inputsPerCharge;
+                return acceptorComp.completedInputs.size >= processorComp.inputsPerCharge;
```

Comment:
This will be tricky when not using a `Map`, please don't use `Object.keys`

Reply:
This is fine, as its now using an array